cmd/migration/v0.9.1-v1/v091-domain/vault_repository.go

```diff
@@ -0,0 +1,29 @@
+package v091domain
+
+import (
+	"github.com/sekulicd/badgerhold/v2"
+)
+
+type VaultRepository interface {
+	GetVault() (*Vault, error)
+}
+
+type vaultRepositoryImpl struct {
+	store *badgerhold.Store
+}
+
+func NewVaultRepositoryImpl(store *badgerhold.Store) VaultRepository {
+	return &vaultRepositoryImpl{store}
+}
+
+func (v *vaultRepositoryImpl) GetVault() (*Vault, error) {
+	var vault Vault
+	if err := v.store.Get(vaultKey, &vault); err != nil {
+		if err == badgerhold.ErrNotFound {
+			return nil, nil
+		}
```

Comment:
either we remove this or we return a `vault not found` error because we want the vault to be there.. if not let's notify it.